Develop3

larkova opened the pull request:
Протестировано веб-приложение Stellar Burgers.
Написаны тесты на следующий функционал:
Регистрация;
Вход;
Переход в ЛК;
Переход из ЛК в Конструктор;
Выход из аккаунта;
Раздел "Конструктор"

larkova committed:
add allure report

larkova committed:
Протестировано веб-приложение Stellar Burgers.
Протестирован следующий функционал:
Регистрация;
Вход;
Переход в ЛК;
Переход из ЛК в Конструктор;
Выход из аккаунта;
Раздел "Конструктор"

pomegranate282922 commented on src/main/java/ru/yandex/praktikum/model/DataForRegistrationGenerator.java:
```diff
@@ -0,0 +1,21 @@
+
+public class DataForRegistrationGenerator {
+    public static DataForRegistration getRandomCorrectData () {
+        String name = RandomStringUtils.randomAlphabetic(10);
```
Отлично, что данные рандомизированы.

pomegranate282922 commented on src/main/java/ru/yandex/praktikum/model/LogInPage.java:
```diff
@@ -0,0 +1,57 @@
+    }
+    //метод открытия страницы
+    public void openPage() {
+        driver.get("https://stellarburgers.nomoreparties.site/login");
```
Все урлы выносим как константы. Нужно исправить.

pomegranate282922 commented on src/main/java/ru/yandex/praktikum/model/LogInPage.java:
```diff
@@ -0,0 +1,57 @@
+    public LogInPage (WebDriver driver) {
+        this.driver = driver;
+    }
+    //метод открытия страницы
```
Сначала в классе перечисляем все элементы, потом все методы. Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/praktikum/model/LogInPage.java:
```diff
@@ -0,0 +1,57 @@
+    private By passwordInput = By.xpath(".//div/input[@name='Пароль']");
+
+    // Локатор кнопки Войти
+    private By LogInButton = By.xpath(".//button[text()='Войти']");
```
Переменные всегда именуем начиная со строчной буквы. Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/praktikum/model/PasswordRecoveryPage.java:
```diff
@@ -0,0 +1,33 @@
+    // метод ожидания загрузки страницы
+    public void waitForOpenPage() {
+        new WebDriverWait(driver, 5)
+                .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id=\"root\"]/div/main/div/div/p/a")));
```
Лучше вынести это в поле класса и переиспользовать.

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/ConstructorTest.java:
```diff
@@ -0,0 +1,68 @@
+        main.clickSaucesPointer();
+        main.clickBunsPointer();
+        main.getAttributeBunsPointerParent();
+        boolean isAttributeContainsCurrent = main.getAttributeBunsPointerParent().contains("current");
```
Все-таки имя класса это часть page object model- лучше спрятать это внутрь po. Можно улучшить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/ConstructorTest.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Test
+    @DisplayName("Проверка перехода в раздел начинки")
+    @Description("Позитивный кейс проверки перехода в раздел начинок на главной странице")
```
Отлично, что даны подробные описания

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/LogInTest.java:
```diff
@@ -0,0 +1,144 @@
+        personalAccount.waitForOpenPage();
+
+        Assert.assertTrue("Не произошел переход в ЛК", personalAccount.isLogoutButtonDisplayed());
+    }
```
Здесь и везде в тестах нужно немного переделать. Для регистрации юзера нужно использовать апи. Затем в конце каждого теста созданный юзер должен быть удалён через апи.
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/RegistrationTest.java:
```diff
@@ -0,0 +1,71 @@
+        registration.clickRegistrationButton();
+
+        Assert.assertTrue("Сообщение о некорректном пароле не отображается", registration.isIncorrectPasswordTitleDisplayed());
+    }
```
Здесь тоже нужно обязательно удалить созданного юзера. Нужно предусмотреть удаление и в негативном кейсе его создания.

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты на месте, есть рандомизация данных, хорошая page object model.
Есть несколько замечаний. Прошу их поправить.

larkova committed:
add allure report

larkova committed:
Выполнены изменения:
URL вынесены в константы;
В классах перечислены все элементы, потом все методы;
Исправлена опечатка в названии переменной (переменные всегда именуем, начиная со строчной буквы);
Для регистрации юзера везде использованы апи;
Каждый созданный юзер удаляется.

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/RegistrationTest.java:
```diff
@@ -0,0 +1,92 @@
+        // создать пользователя
+        userClient = new UserClient();
+        User user = UserGenerator.getRandom();
+        userClient.createUser(user);
```
Здесь есть ошибочка. Ты создаёшь юзера через апи и его же удаляешь. Зачем? Это же тест на регистрацию - то есть ты создаёшь юзера в самом тесте. И именно его и нужно удалить.
Также нужно удалять и юзера, который может ( гипотетически ) создасться во втором тесте.
Также есть другая проблема: ты создала юзера в before, создаёшь такого же в самом тесте, но он почему-то не падает, хотя должен. Перепроверь почему и исправь, пожалуйста

pomegranate282922 reviewed (commented):
Все хорошо, кроме момента с тестом на регистрацию. Там надо ещё поразбираться.

larkova committed:
Исправлены тесты на регистрацию пользователя:
Пользователь создается через форму регистрации и удаляется через API как в позитивном сценарии, так и в негативном (если в негативном сценарии вдруг создастся пользователь)

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/RegistrationTest.java:
```diff
@@ -58,35 +54,44 @@ public void checkRegistrationWithCorrectData() {
         registration.clickRegistrationButton();
 
         Assert.assertTrue("Сообщение о некорректном пароле не отображается", registration.isIncorrectPasswordTitleDisplayed());
+
```
Сюда при успешном создании заказа код не дойдёт, потому что тест упадёт на ассерте раньше. Можно улучшить

pomegranate282922 reviewed (approved):
Осталось небольшое замечание, но считаю его некритичным.
